code/renderers/vue3/src/docs/sourceDecorator.ts

```diff
@@ -0,0 +1,275 @@
+/* eslint-disable no-underscore-dangle */
+import { addons, useEffect } from '@storybook/preview-api';
+import type { ArgTypes, Args, StoryContext, Renderer } from '@storybook/types';
+
+import { SourceType, SNIPPET_RENDERED } from '@storybook/docs-tools';
+
+import { format } from 'prettier';
```

Comment:
I'm worried about the bundle-size implications of this. Isn't prettier like 1MB?

Reply:
What happens if we do not format the code, and let the syntax-highlighter showing the code do this instead?
It has prettier build-in as well. But the problem might be it't not setup for vue code?

Reply:
i know you would comment on this, it is more for indentention, my initial code was without that but it was not really pretty

Reply:
i know you would comment on this, it is more for indentention, my initial code was without that but it was not really pretty

Reply:
but it is already bundled  i guess, and i use it for parsing code as well to get the template from the StoryFn

Reply:
I'm handling beautifully  complex nested components 
<img width="1172" alt="image" src="https://user-images.githubusercontent.com/711292/212149970-7c5add80-3598-442e-8bf2-608cac01fdc6.png">

Reply:
@shilman  has added prettier before 
<img width="1121" alt="image" src="https://user-images.githubusercontent.com/711292/212152280-325c77fe-e36c-40dc-ad44-a78fe7883d0b.png">

Reply:
i have new commit to support multiple and nested components with v-bind and dynamic and static args in the same story 

<img width="1187" alt="image" src="https://user-images.githubusercontent.com/711292/212182262-55b91983-4a58-4ea4-868f-9dc0e67b5a34.png">

Story source is like this 

<img width="739" alt="image" src="https://user-images.githubusercontent.com/711292/212182376-abe362eb-51f5-4716-8554-d778e2673728.png">

Reply:
👍